community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/RecoveryVisitor.java

```diff
@@ -74,6 +75,7 @@ public boolean visit( RecoverableTransaction transaction ) throws IOException
         }
 
         lastTransactionIdApplied = txId;
+        lastTransactionCommitTimestamp = transaction.representation().getCommitEntry().getTimeWritten();
```

Comment:
Since the grace period is only a heuristic, I suppose it doesn't matter if concurrent transactions are out of order in the log.